Interop/StubGenerator/src/main/kotlin/org/jetbrains/kotlin/native/interop/gen/StubIr.kt

```diff
@@ -113,29 +124,54 @@ interface AnnotationHolder {
     val annotations: List<AnnotationStub>
 }
 
-sealed class AnnotationStub {
-    sealed class ObjC : AnnotationStub() {
-        object ConsumesReceiver : ObjC()
-        object ReturnsRetained : ObjC()
-        class Method(val selector: String, val encoding: String, val isStret: Boolean = false) : ObjC()
-        class Factory(val selector: String, val encoding: String, val isStret: Boolean = false) : ObjC()
-        object Consumed : ObjC()
-        class Constructor(val selector: String, val designated: Boolean) : ObjC()
-        class ExternalClass(val protocolGetter: String = "", val binaryName: String = "") : ObjC()
+sealed class AnnotationStub(val classifier: Classifier) {
```

Comment:
Could the classifier be reused when rendering stubs to text?

Reply:
It will be. But I think it better be done in the separate PR since this one is already quite big.